notebooks/concepts/imviz_advanced_aper_phot.ipynb

```diff
@@ -0,0 +1,403 @@
+   "source": [
+    "# imviz.link_data(link_type='wcs')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "987c96f0-ba8b-42b1-9ffd-6f50db3f190c",
+   "metadata": {},
+   "source": [
+    "### Image 2.1: Add rotation to downsampled image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "91dc1847-1ae1-4a51-aa44-feac12f4c290",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Like above but with extra rotation\n",
+    "celestial_rotation =  models.RotateNative2Celestial(\n",
+    "    197.8925 * u.deg, -1.36555556 * u.deg, 190 * u.deg)\n",
+    "\n",
+    "det2sky = shift_by_crpix | rotation | tan | celestial_rotation\n",
+    "det2sky.name = \"linear_transform\"\n",
+    "\n",
+    "pipeline = [(detector_frame, det2sky),\n",
+    "            (sky_frame, None)]\n",
+    "gw_rot = GWCS(pipeline)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "43797cca-c6e1-4bc0-ad66-46ba5b16bcfe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This takes much longer when GWCS has distortion.\n",
+    "data_rot, _ = reproject_interp(image_2, gw_rot, shape_out=data_sm.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1b52e436-2cd3-4a42-b7a8-d489a6fb06ad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_2_1 = NDData(data_rot, wcs=gw_rot)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42cadf07-99cc-4816-b6c3-43847c73478e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imviz.load_data(image_2_1, data_label=\"gwcs_rot\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d78fcfe-d224-410a-a9d8-7ff84a3bd660",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imviz.link_data(link_type='wcs')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe16d7c9-5b41-45fe-b7a5-3f6c1e91ec95",
+   "metadata": {},
+   "source": [
+    "## How is glue projecting the rectangle?\n",
+    "\n",
+    "How is `glue` really projecting the rectangle for all the data?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "69549536-96f5-43ea-afc4-aaa6f79723ba",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rect_grp = imviz.app.data_collection.subset_groups[3]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5958db68-2d75-49da-bac5-4a1eefb59ded",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axs = plt.subplots(1, 3)\n",
+    "axs[0].imshow(rect_grp.subsets[0].to_mask(), origin='lower')\n",
+    "axs[0].set_xlim(210, 250)\n",
+    "axs[0].set_ylim(140, 160)\n",
+    "axs[1].imshow(rect_grp.subsets[1].to_mask(), origin='lower')\n",
+    "axs[1].set_xlim(105, 125)\n",
+    "axs[1].set_ylim(70, 80)\n",
+    "axs[2].imshow(rect_grp.subsets[2].to_mask(), origin='lower')\n",
+    "axs[2].set_xlim(105, 125)\n",
+    "axs[2].set_ylim(70, 80);"
```

Comment:
This would give:

![Screenshot 2023-04-14 173003](https://user-images.githubusercontent.com/2090236/232159733-8f5fffbf-07df-40c9-aaf1-388ca35a7df0.jpg)

This corresponds to the following images loaded into Imviz and linked by WCS:

![Screenshot 2023-04-14 173442](https://user-images.githubusercontent.com/2090236/232159776-502d2844-d7ab-4f68-b7c7-8b0b5e4bccf7.jpg)

Reply:
I don't know how to:

* pass in arbitrary mask shape (and maybe its bounding box if there is a way to extract mask without reading in the whole image) into `photutils.aperture.ApertureStats` -- @larrybradley ?
* extract subset mask in glue without reading in all the data into memory -- @astrofrog ?

Reply:
@larrybradley , to answer your question, if I draw a rectangle on the third image (down-sampled and rotated), the rectangle comes back warped as if I drew it on the reference image, which is what I suspected would happen.

![Screenshot 2023-04-17 172540](https://user-images.githubusercontent.com/2090236/232614685-b6a18beb-7af0-427c-bcac-3245e9918e1d.jpg)